exporter/awsemfexporter/datapoint.go

```diff
@@ -18,25 +18,26 @@ import (
 	"time"
 
 	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/otel/attribute"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws"
 )
 
-var rateMetricCalculator = newFloat64RateCalculator()
-
-func newFloat64RateCalculator() aws.MetricCalculator {
-	return aws.NewMetricCalculator(func(prev *aws.MetricValue, val interface{}, timestamp time.Time) (interface{}, bool) {
-		if prev != nil {
-			deltaTimestampMs := timestamp.Sub(prev.Timestamp).Milliseconds()
-			deltaValue := val.(float64) - prev.RawValue.(float64)
-			if deltaTimestampMs > 50*time.Millisecond.Milliseconds() && deltaValue >= 0 {
-				return deltaValue * 1e3 / float64(deltaTimestampMs), true
-			}
-		}
-		return float64(0), true
-	})
+var deltaMetricCalculator = aws.NewFloat64DeltaCalculator()
+var summaryMetricCalculator = aws.NewMetricCalculator(calculateSummaryDelta)
+
+func calculateSummaryDelta(prev *aws.MetricValue, val interface{}, timestampMs time.Time) (interface{}, bool) {
+	sumCount := val.([]interface{})
```

Comment:
Can this be a struct instead of a dynamic array?

Reply:
Done.